addons/portal/models/portal_mixin.py

```diff
@@ -76,3 +102,43 @@ def _notify_get_groups(self, message, groups):
         else:
             new_group = []
         return super(PortalMixin, self)._notify_get_groups(message, new_group + groups)
+
+    @api.multi
+    def get_access_action(self, access_uid=None):
+        """ Instead of the classic form view, redirect to the online document for
+        portal users or if force_website=True in the context. """
+        # TDE note: read access on document to portal users granted to followed sales orders
+        self.ensure_one()
+
+        user, record = self.env.user, self
+        if access_uid:
```

Comment:
I know this is code from standard, but I would suggest an improvement for security purpose (we will probably have to backport it): before sudo-ing as access_uid we should ensure current user has a read access on the document itself.

Method will be something like
```
user, record = self.env.user, self
if access_uid:
    try:
        record.check_access_rule("read")
    except AccessError:
        return super()
    user = ...
    record = ...
.. remaining of method where we keep the second check access rule
```